django/contrib/admin/options.py

```diff
@@ -962,12 +962,25 @@ def log_deletion(self, request, obj, object_repr):
             action_flag=DELETION,
         )
 
-    @display(description=mark_safe('<input type="checkbox" id="action-toggle">'))
+    @display(
+        description=mark_safe(
+            gettext_lazy(
+                '<input type="checkbox" id="action-toggle"'
+                'aria-label="Select all objects on this page for an action">'
+            )
+        )
+    )
```

Comment:
We could move the description to `result_headers()` to avoid adding raw HTML to translations:
```diff
diff --git a/django/contrib/admin/options.py b/django/contrib/admin/options.py
index ed70f93628..34ea5ce687 100644
--- a/django/contrib/admin/options.py
+++ b/django/contrib/admin/options.py
@@ -962,7 +962,6 @@ class ModelAdmin(BaseModelAdmin):
             action_flag=DELETION,
         )
 
-    @display(description=mark_safe('<input type="checkbox" id="action-toggle">'))
     def action_checkbox(self, obj):
         """
         A list_display column containing a checkbox widget.
diff --git a/django/contrib/admin/templatetags/admin_list.py b/django/contrib/admin/templatetags/admin_list.py
index ee6f3a7666..761a64ae5f 100644
--- a/django/contrib/admin/templatetags/admin_list.py
+++ b/django/contrib/admin/templatetags/admin_list.py
@@ -96,8 +96,12 @@ def result_headers(cl):
 
             # if the field is the action checkbox: no sorting and special class
             if field_name == "action_checkbox":
+                aria_label = _("Select all objects on this page for an action")
                 yield {
-                    "text": text,
+                    "text": mark_safe(
+                        f'<input type="checkbox" id="action-toggle" '
+                        f'aria-label="{aria_label}">'
+                    ),
                     "class_attrib": mark_safe(' class="action-checkbox-column"'),
                     "sortable": False,
                 }
```